src/Wallabag/CoreBundle/Repository/EntryRepository.php

```diff
@@ -368,6 +368,7 @@ public function findByHashedUrlAndUserId($hashedUrl, $userId)
     {
         $res = $this->createQueryBuilder('e')
             ->where('e.hashedUrl = :hashed_url')->setParameter('hashed_url', $hashedUrl)
+            ->orWhere('e.hashedGivenUrl = :hashed_given_url')->setParameter('hashed_given_url', $hashedUrl)
             ->andWhere('e.user = :user_id')->setParameter('user_id', $userId)
```

Comment:
I can't remember if Doctrine respects the order of `WHERE` statements ? I think it may be faster if we have the condition on `user_id` first.

Reply:
This code generates this query:

```sql
SELECT e FROM Wallabag\CoreBundle\Entity\Entry e WHERE (e.hashedUrl = :hashed_url OR e.hashedGivenUrl = :hashed_given_url) AND e.user = :user_id
```
And I can't find a way to handle the `user_id` first and then the `OR` without _almost_ write the query by hand.
Also there is an index on each field of the query, it should fast enough

Reply:
But the index is not used because we use a `OR`.
I'll split the query in two and create two indexes so they'll be used